pkg/validation/internal/multiarch.go

```diff
@@ -325,21 +403,27 @@ func (data *multiArchValidator) doChecks() {
 	// on the check above. The following check raise a warning when it is possible to check that the Operator
 	// manager image(s) supports architecture(s) not defined via labels. Therefore, it shows like the labels are missing
 	data.checkMissingLabelsForArchs()
-	data.checkMissingLabelsForSO()
+	data.checkMissingLabelsForOS()
 	// the following check will raise warnings when is possible to verify that the images defined in the CSV
 	// does not provide the same architecture(s) supported by the Operator manager or defined via the labels
-	data.checkMissingSupportForOtherImages()
+	data.checkMissingSupportForOtherImages(data.otherCSVDeploymentImages)
+	data.checkMissingSupportForOtherImages(data.relatedImages)
+
+	// the following check will raise warnings when nodeAffinity isn't set to ensure that the pod spec will only
+	// target nodes matching the platforms (os/arch) specified in the manifest.
+	data.checkNodeAffinity(data.managerImages)
+	data.checkNodeAffinity(data.otherCSVDeploymentImages)
 }
 
 // checkMissingSupportForOtherImages checks if any image is missing some arch or so found
-// (probably the image should support the arch or SO )
-func (data *multiArchValidator) checkMissingSupportForOtherImages() {
-	for image, plaformFromImage := range data.allOtherImages {
+// (probably the image should support the arch or OS )
+func (data *multiArchValidator) checkMissingSupportForOtherImages(images map[string][]platform) {
+	for image, platformFromImage := range images {
```

Comment:
Note: No functional changes were made in this function other than the new argument.